### ISSUE TITLE
[stable/nginx-ingress] Updating the chart without changing nginx image version break the ingress

### ISSUE THREAD

**Issue by pierluigilenoci** [helm/charts#16809]
**Describe the bug**
If you update the chart without changing the nginx image the pods stop to working.

**Version of Helm and Kubernetes**:
K8s v1.12.7
Helm v2.14.3

**Which chart**:
v1.17.1

**What happened**:
I had nginx-ingress installed (nginx-ingress-1.15.0, v0.24.1).
I updated the chart (nginx-ingress-1.17.1, v0.24.1).
The nginx version was not updated by choice (due to this problem https://github.com/kubernetes/ingress-nginx/issues/4305).
The nginx pods have begun to generate these errors:
```
E0902 08:07:22.820980      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:156: Failed to list *v1.Pod: Unauthorized
E0902 08:07:22.821894      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:154: Failed to list *v1.Secret: Unauthorized
E0902 08:07:22.822561      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:153: Failed to list *v1.Service: Unauthorized
E0902 08:07:23.118202      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:176: Failed to list *v1beta1.Ingress: Unauthorized
E0902 08:07:23.278460      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:152: Failed to list *v1.Endpoints: Unauthorized
E0902 08:07:23.563842      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:155: Failed to list *v1.ConfigMap: Unauthorized
E0902 08:07:23.824365      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:156: Failed to list *v1.Pod: Unauthorized
E0902 08:07:23.824869      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:154: Failed to list *v1.Secret: Unauthorized
E0902 08:07:23.826023      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:153: Failed to list *v1.Service: Unauthorized
E0902 08:07:24.121045      12 reflector.go:126] k8s.io/ingress-nginx/internal/ingress/controller/store/store.go:176: Failed to list *v1beta1.Ingress: Unauthorized
```

And then I found this error in the pods:
```
MountVolume.SetUp failed for volume "nginx-ingress-token-2nf6f" : secrets "nginx-ingress-token-2nf6f" not found
```

Basically the chart update changed the name of the secret. The pods had the old and stopped working. By restarting them they took the secret correctly and restarted to work properly. 

**What you expected to happen**:
If I update only the chart I would like the pods to continue to work

**How to reproduce it** (as minimally and precisely as possible):
Redo the same step that I did.

**Anything else we need to know**:

**Comment by PierluigiLenociAkelius** [comment]
@ChiefAlexander @taharah could you please take a look?

**Comment by pierluigilenoci** [comment]
Toc toc

**Comment by pierluigilenoci** [comment]
@ChiefAlexander @taharah can I have a feedback?

**Comment by pierluigilenoci** [comment]
I was in vacation... :( @ChiefAlexander @taharah
